### PR TITLE
change default docker image repository from docker.io to quay.io

### PR DESCRIPTION
**Pull request by aanm**
With the recent limitations introduced by Docker, docker hub has been
rate limiting docker pulls on images for non-registered users. To avoid
this limitation we should switch the default container image
repositories to quay.io.

Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Review by borkmann — APPROVED** [review]

**Review by fristonio — APPROVED** [review]
LGTM :rocket: 
One question regarding this in general though, we have a bunch of cilium images we use in our test suites(eg cilium/json-mock) and we still use docker hub for them. While the frequency of our tests will be less than the rate limit imposed by docker should we also consider moving them to quay. Ideally, I could think dev images like this to have a separate account so we don't bring in a lot of noise to the cilium quay account. Any thoughts around this?

**Review by michi-covalent — APPROVED** [review]

**Comment by joestringer** [comment]
> Ideally, I could think dev images like this to have a separate account so we don't bring in a lot of noise to the cilium quay account.

FWIW we have applied for a dockerhub OSS account which should alleviate these types of issues. I do not know how long it will take for them to get back to us but I will be sure to update everyone if I hear anything back.

**Review by joestringer — APPROVED** [review]
Here goes...

**Review by christarazi — APPROVED** [review]

File: Makefile.docker [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 UTC_DATE=$(shell date -u "+%Y-%m-%d")
+REPOSITORY ?= quay.io
```
Comment (sayboras) [inline Makefile.docker]: while this one is making sense,I can't find anywhere where REPOSITORY is passed as docker.io, will cilium publish images to docker.io anymore ?

Comment (sayboras) [inline Makefile.docker]: nit: contrib/scripts/push_manifest.sh script is using term `DOCKER_REGISTRY` instead, so just small nitpick for consistency.

**Review by sayboras — APPROVED** [review]
LGTM.

Minor points below.

**Review by kkourt — APPROVED** [review]

File: Makefile.docker (hunk 1 above)
Comment (aanm) [reply Makefile.docker]: @sayboras the images aren't "pushed", both docker and quay have rules to automatically build images once a tag is released. The GH actions that we have are currently doing "push" to both `quay.io` and `docker.io` but to the `*-dev` repositories.

Comment (aanm) [reply Makefile.docker]: good point, will switch.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
The CI is passing for the provisioning of the images so we can merge the PR now.

**Pull request merged**